### PR TITLE
Tao 6860/clean out all the file statements linked to non existent files

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/TAO-6860

Cleans additionally to specified in story

Files that no more stored as per oat-sa/extension-tao-item#327

### CODE REVIEW

File: common/report/class.Report.php [hunk 1]
```diff
@@ -337,4 +337,13 @@ public function toArray()
             }, $this->elements)
         ]);
     }
+
+    /**
+     * @return int
+     */
+    public function count()
```
Comment (zagovorichev) [inline common/report/class.Report.php]: do you really need this?

File: scripts/tools/CleanUpOrphanFiles.php [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        }
+        $this->getLogger()->info($message);
+        $this->remove($resource);
+    }
+
+    /**
+     * @param core_kernel_classes_Resource $resource
+     * @param FileSystemHandler $file
+     */
+    protected function manageOrpane(core_kernel_classes_Resource $resource, FileSystemHandler $file)
```
Comment (zagovorichev) [inline scripts/tools/CleanUpOrphanFiles.php]: `manageOrphan`?

**Review by zagovorichev — CHANGES_REQUESTED** [review]

**Review by zagovorichev — APPROVED** [review]

**Comment by siwane** [comment]
@martijn-tao as you have done same kind of script, could u try to harmonize both?

**Comment by martijn-tao** [comment]
> @martijn-tao as you have done same kind of script, could u try to harmonize both?

For my script we've introduced a ResourceCollection which could be used here as well. 
It might also be a good idea to split some functionality to make the script slightly less complex.

More details in my PR: https://github.com/oat-sa/generis/pull/543

**Comment by boajer** [comment]
@martijn-tao I really like your ResourceCollection. Looks like https://github.com/oat-sa/generis/pull/543 should be stabilised and merged first, before redoing my script.

**Comment by martijn-tao** [comment]
> @martijn-tao I really like your ResourceCollection. Looks like #543 should be stabilised and merged first, before redoing my script.

@boajer Thanks! Sounds like a good idea. That PR should almost be done _(I believe)_. The current ResourceCollection shouldn't change anymore, so you could already pull that, and work on a refactor using it in the meantime.

**Comment by martijn-tao** [comment]
@boajer My work got merged, so the ResourceCollection is now available 🎉

**Comment by bartlomiejmarszal** [comment]
I fix only part of this script that was responsible for using different service to collect statements that were not available in filesystem

**Review by ekkinox — APPROVED** [review]

File: scripts/tools/CleanUpOrphanFiles.php [hunk 3]
```diff
@@ -168,10 +168,8 @@ private function getRedundantFiles()
      */
     private function markForRemoval(core_kernel_classes_Resource $resource)
     {
-        if ($this->wetRun) {
```
Comment (boajer) [inline scripts/tools/CleanUpOrphanFiles.php]: Why you've removed wet\dry run support?

**Review by tikhanovichA — APPROVED** [review]

File: scripts/tools/CleanUpOrphanFiles.php [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    private $wetRun = false;
+    private $verbose = false;
+    private $removedCount = 0;
+    private $redundantCount = 0;
+    private $affectedCount = 0;
+    private $errorsCount = 0;
+    private $report;
+    private $markedTobeRemoved =[];
+
+    protected function provideOptions()
```
Comment (martijn-tao) [inline scripts/tools/CleanUpOrphanFiles.php]: Some DocBlocks are missing

File: scripts/tools/CleanUpOrphanFiles.php [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+    {
+        return $this->getServiceLocator()
+            ->get(\common_persistence_Manager::SERVICE_ID)
+            ->getPersistenceById('default');
+    }
+
+    private function getRedundantFiles()
+    {
+        return [
+            'qti.xml' //special case, see linked story ( has been stored at RDS, but never referenced via resource ).
```
Comment (martijn-tao) [inline scripts/tools/CleanUpOrphanFiles.php]: If this method had a docblock, that would be a perfect place for this comment ;)

**Review by martijn-tao — CHANGES_REQUESTED** [review]

File: scripts/tools/CleanUpOrphanFiles.php [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+        return true;
+    }
+
+    /**
+     * @param core_kernel_classes_Resource $resource
+     * @return bool
+     */
+    private function isOrphan(core_kernel_classes_Resource $resource)
+    {
+        $sql = 'SELECT subject FROM statements s WHERE s.object=?';
```
Comment (siarhei-baradzin) [inline scripts/tools/CleanUpOrphanFiles.php]: `SELECT 1 ... LIMIT 1` could improve performance if it matters here

**Review by siarhei-baradzin — APPROVED** [review]

File: scripts/tools/CleanUpOrphanFiles.php (hunk 4 above)
Comment (bartlomiejmarszal) [reply scripts/tools/CleanUpOrphanFiles.php]: We are going to remove this script anyway. Is there a point to add docblocks now?

File: scripts/tools/CleanUpOrphanFiles.php (hunk 6 above)
Comment (bartlomiejmarszal) [reply scripts/tools/CleanUpOrphanFiles.php]: This is a one-time action. I don't think that this is important here.

File: scripts/tools/CleanUpOrphanFiles.php [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+    }
+
+    private function getPersistence()
+    {
+        return $this->getServiceLocator()
+            ->get(\common_persistence_Manager::SERVICE_ID)
+            ->getPersistenceById('default');
+    }
+
+    private function getRedundantFiles()
```
Comment (siarhei-baradzin) [inline scripts/tools/CleanUpOrphanFiles.php]: Pls pay attention this method isn't used, ignore my notice if it's ok (I'm not aware of the whole task)

**Review by martijn-tao — APPROVED** [review]

**Comment by martijn-tao** [comment]
🙈

**Pull request merged**